oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/MetricsEsDAO.java

```diff
@@ -52,50 +52,47 @@ protected MetricsEsDAO(ElasticSearchClient client,
-                                 // as a new day comes, and the index cache is  pseudo real time.
-                                 // This case doesn't affect the result, just has lower performance due to using the alias name.
-                                 // Another case is that a removed index showing existing also due to latency,
-                                 // which could cause multiGet fails
-                                 // but this should not happen in the real runtime, TTL timer only removed the oldest indices,
-                                 // which should not have an update/insert.
                                  String indexName = TimeSeriesUtils.writeIndexName(model, metric.getTimeBucket());
-                                 // Format the name to follow the global physical index naming policy.
-                                 if (!IndicesMetadataCache.INSTANCE.isExisting(
-                                     getClient().formatIndexName(indexName))) {
-                                     indexName = IndexController.INSTANCE.getTableName(model);
-                                 }
-                                 return indexName;
+                                 groupIndices.computeIfAbsent(indexName, v -> new ArrayList<>()).add(metric);
                              } else {
                                  // Metadata level metrics, always use alias name, due to the physical index of the records
                                  // can't be located through timestamp.
-                                 return IndexController.INSTANCE.getTableName(model);
+                                 String indexName = IndexController.INSTANCE.getTableName(model);
+                                 groupAliasIndices.computeIfAbsent(indexName, v -> new ArrayList<>()).add(metric);
                              }
-                         })
-                     );
-
+                         });
         // The groupIndices mostly include one or two group,
         // the current day and the T-1 day(if at the edge between days)
         List<Metrics> result = new ArrayList<>(metrics.size());
-        groupIndices.forEach((tableName, metricList) -> {
+        groupAliasIndices.forEach((tableName, metricList) -> {
             List<String> ids = metricList.stream()
                                          .map(item -> IndexController.INSTANCE.generateDocId(model, item.id()))
                                          .collect(Collectors.toList());
-            final SearchResponse response = getClient().ids(tableName, ids);
+            final SearchResponse response = getClient().idsWithIndexAlias(tableName, ids);
             response.getHits().getHits().forEach(hit -> {
                 Metrics source = storageBuilder.storage2Entity(new HashMapConverter.ToEntity(hit.getSource()));
                 result.add(source);
             });
         });
 
+        Map<String, List<String>> indexIdsGroup = new HashMap<>();
```

Comment:
I think we can merge `groupAliasIndices` into `groupIndices` and call `getClient().ids` once, then remove `idsWithIndexAlias`.

Reply:
`_mget` doesn't support alias if there multi indexes... Do you have a better idea?

Reply:
We don't have to use `mget`. Like https://github.com/apache/skywalking/pull/9339#discussion_r919702835 mentioned, we could keep alias.

Reply:
> `_mget` doesn't support alias if there multi indexes... Do you have a better idea?

Well. Then renaming the method to what @wu-sheng suggest is ok to me